src/openrct2/ride/ride.c

```diff
@@ -6970,26 +7056,34 @@ static sint32 ride_get_track_length(Ride *ride)
             if (mapElement->base_height != z)
                 continue;
 
-            goto foundTrack;
-        } while (!map_element_is_last_for_tile(mapElement++));
+            foundTrack = true;
+        }
+        while (!map_element_is_last_for_tile(mapElement++) && !foundTrack);
```

Comment:
This is not correct. Previous code would skip over incrementing `mapElement` if a track was found. `mapElement` is used later on, so it needs to be fixed. Change the order of conditions here to make it match.